common/src/View/Actions.h

```diff
@@ -105,6 +96,47 @@ namespace TrenchBroom {
             Action& operator=(Action&& other) = default;
         };
 
+        /**
+         * ExecuteFn has type ActionExecutionContext& -> void
+         * EnabledFn has type ActionExecutionContext& -> bool
+         * CheckedFn has type ActionExecutionContext& -> bool
+         */
+        template <class ExecuteFn, class EnabledFn, class CheckedFn>
```

Comment:
This is a subclass purely to avoid having all the code from `Action` in the header file?

Reply:
They need to be stored in a vector, so they need to be a subclass since the class size changes depending on what you captue

Reply:
But you're allocating on the heap, so the size shouldn't matter? But I assume the reason is that you to be able to treat the different template instantiations polymorphically, which is why Action cannot be a template.

Reply:
Yeah that's that I meant :)